BUG: When floating border is toggled off, can't be toggled on again
--- dwm.c.orig	2015-12-31 15:27:32.259517555 -0800
+++ dwm.c	2015-12-31 16:36:41.106570362 -0800
@@ -94,7 +94,7 @@ struct Client {
 	int basew, baseh, incw, inch, maxw, maxh, minw, minh;
 	int bw, oldbw;
 	unsigned int tags;
-	int ismax, wasfloating, isfixed, isfloating, isurgent, neverfocus, oldstate, isfullscreen, iscentered;
+	int ismax, wasfloating, isfixed, isfloating, isurgent, neverfocus, oldstate, isfullscreen, iscentered, hasborders;
 	Client *next;
 	Client *snext;
 	Monitor *mon;
@@ -218,6 +218,7 @@ static void tagmon(const Arg *arg);
 static void tile(Monitor *);
 static void togglebar(const Arg *arg);
 static void togglefloating(const Arg *arg);
+static void toggleborders(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
 static void unfocus(Client *c, int setfocus);
@@ -1832,6 +1833,11 @@ tile(Monitor *m)
 	else
 		mw = m->ww;
 	for (i = my = ty = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
+		/* if(!c->hasborders) */
+		/* { */
+		/* 	c->oldbw = c->bw; */
+		/* 	c->bw = 0; */
+		/* } */
 		if (i < m->nmaster) {
 			h = (m->wh - my) / (MIN(n, m->nmaster) - i);
 			if(n == 1)
@@ -1884,6 +1890,29 @@ togglefloating(const Arg *arg)
 }
 
 void
+toggleborders(const Arg *arg)
+{
+	Client *c = selmon->sel;
+
+	if (!selmon->sel)
+		return;
+	if (selmon->sel->isfullscreen) /* fullscreen windows shouldn't have borders */
+		return;
+	selmon->sel->hasborders = !selmon->sel->hasborders;
+	if(selmon->sel->hasborders)
+	{
+		selmon->sel->bw = selmon->sel->oldbw;
+		selmon->sel->oldbw = 0;
+	}
+	else
+	{
+		selmon->sel->oldbw = selmon->sel->bw;
+		selmon->sel->bw = 0;
+	}
+	resizeclient(c, c->x, c->y, c->w, c->h);
+}
+
+void
 toggletag(const Arg *arg)
 {
 	unsigned int newtags;
